src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -461,6 +461,15 @@ bool Profile::HasIcon() const noexcept
     return _icon.has_value();
 }
 
+// Method Description:
+// - Sets this profile's icon path.
+// Arguments:
+// - path: the path
+void Profile::SetIconPath(const std::wstring_view& path) noexcept
+{

```

Comment:
Should ` void SetIconPath(std::wstring_view path) noexcept`. 

>https://abseil.io/docs/cpp/guides/strings
>string_view objects are very lightweight, so you should always pass them by value within your methods and functions; don’t pass a const absl::string_view &. (Passing absl::string_view instead of const absl::string_view & has the same algorithmic complexity, but because of register allocation and parameter passing rules, it is generally faster to pass by value in this case.)

`const std::wstring_view &`：
```asm
movq	(%rdi), %rax
```

`std::wstring_view`:
```asm
movq	%rdi, %rax
```

Reply:
I did not know that, that's fascinating. @adiviness would be interested also

Reply:
@fcharlie thanks for the info! that's really good to know.

Reply:
I wonder if CppCoreChecks or linting could catch this...

Reply:
Sidenote that this seems to match the advice in the ISO C++ guidelines:
[F.16: For “in” parameters, pass cheaply-copied types by value and others by reference to const](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const)

Key points are:
> What is “cheap to copy” depends on the machine architecture, but two or three words (doubles, pointers, references) are usually best passed by value. When copying is cheap, nothing beats the simplicity and safety of copying, and for small objects (up to two or three words) it is also faster than passing by reference because it does not require an extra indirection to access from the function.

However, using by value instead of by const reference is only appropriate because string_view is readonly.  const& is preferred when declaring read-only intent or when there's a question around lifetime management.

Reply:
Is there a difference between `std::wstring_view` and `const std::wstring_view`? I realize that the string views are read only but I'm curious if the compiler is aware enough to take advantage of it if possible.

Reply:
There's an argument for using const by-value string views (in that the data pointer is made `const` as well)

Reply:
The std::string_view's data pointer is [always const](https://en.cppreference.com/w/cpp/string/basic_string_view/data):
`constexpr const_pointer data() const noexcept;`

This is different than gsl's string_view.

Reply:
> Is there a difference between `std::wstring_view` and `const std::wstring_view`? I realize that the string views are read only but I'm curious if the compiler is aware enough to take advantage of it if possible.

In general, compilers don't usually take advantage of const, per LLVM's Chandler Carruth.  This is covered in Chandler's LLVM Optimization talk (don't have exact timestamp):
https://www.youtube.com/watch?v=FnGCDLhaxKU

const is really more about correctness than optimization.  With that said, I don't know specifically about msvc, but I doubt it.